src/v/pandaproxy/schema_registry/store.h

```diff
@@ -860,18 +867,29 @@ class store {
         return sub_it;
     }
 
-    static result<std::vector<subject_version_entry>::iterator>
+    static result<chunked_vector<subject_version_entry>::iterator>
     get_version_iter(
       subject_map::value_type& sub_entry,
       schema_version version,
       include_deleted inc_del) {
-        const subject_map::value_type& const_entry = sub_entry;
-        return detail::make_non_const_iterator(
-          sub_entry.second.versions,
-          get_version_iter(const_entry, version, inc_del));
+        auto& versions = sub_entry.second.versions;
+        auto v_it = std::lower_bound(
+          versions.begin(),
+          versions.end(),
+          version,
+          [](const subject_version_entry& lhs, schema_version rhs) {
+              return lhs.version < rhs;
+          });
+        if (v_it == versions.end() || v_it->version != version) {
+            return not_found(sub_entry.first, version);
+        }
+        if (!inc_del && v_it->deleted) {
+            return not_found(sub_entry.first, version);
+        }
+        return v_it;
```

Comment:
its not really clear how this change relates to changing the container to chunked vector.